### PR TITLE
Error logs

### PR DESCRIPTION
**Pull request by wangfan860**
### Improvements
1. log endpoint update
2. workflow error interpret update

### CODE REVIEW

File: src/argowrapper/engine/argo_engine.py [hunk 1]
```diff
@@ -122,32 +122,79 @@ def _get_workflow_node_artifact(self, uid: str, node_id: str) -> str:
...
                     {
                         "name": step.get("name"),
-                        "node_id": node_id,
+                        #    "node_id": node_id,
                         "node_type": node_type,
                         "node_phase": node_phase,
                         "step_name": node_step,
                         "step_template": node_step_template,
-                        "error_message": message,
+                        #    "error_message": message,
```
Comment (pieterlukasse) [inline src/argowrapper/engine/argo_engine.py:197]: Ok, this maybe addresses my first comment. So we are not returning the message after all.
Let's remove all commented out lines.

File: src/argowrapper/workflows/argo_workflows/gwas.py [hunk 2]
```diff
@@ -193,17 +193,21 @@ def interpret_gwas_workflow_error(step_name: str, step_log: str) -> str:
...
         of Failed Retry node
         """
-        if step_name == "run-null-model" and "solve.default" in step_log:
-            show_error = "Small cohort size or unbalanced cohort sizes. A cohort size of about 700 or below may fail for continuous outcome workflow."
-        elif step_name == "run-plots" and "mutate" in step_log:
-            show_error = "Small cohort size or unbalanced cohort sizes."
+        if step_name == "run-null-model" and "system is exactly singular" in step_log:
+            show_error = "The error occurred due to small cohort size or unbalanced cohort sizes. Please ensure that the cohorts selected for your analysis are sufficiently large and balanced."
+        # elif step_name == "run-plots" and "mutate" in step_log:
+        #     show_error = "The error occurred due to small cohort size or unbalanced cohort sizes. Please ensure that the cohorts selected for your analysis are sufficiently large and balanced."
```
Comment (pieterlukasse) [inline src/argowrapper/workflows/argo_workflows/gwas.py:199]: please remove these and other commented out lines.

Comment (wangfan860) [reply src/argowrapper/workflows/argo_workflows/gwas.py]: commented out lines were removed.

File: src/argowrapper/engine/argo_engine.py (hunk 1 above)
Comment (wangfan860) [reply src/argowrapper/engine/argo_engine.py]: you are correct. All commented out lines were removed.

File: test/test_argo_engine.py [hunk 3]
```diff
@@ -774,16 +775,17 @@ def test_argo_engine_get_workflow_log_succeeded():
...
-    assert workflow_errors[0]["name"] == "step_one_name"
+    assert workflow_errors[0]["name"] == "step_one_name(0)"
     assert (
         workflow_errors[0]["error_interpreted"]
-        == "Timeout occurred while fetching attrition table information. Please retry."
+        == "A timeout occurred while fetching the attrition table information. Please retry running your workflow."
     )
-    assert workflow_errors[0]["error_message"] == "Error (exit code 126)"
+    # Because we don't want to show detailed error messages to the logs endpoint
+    # assert workflow_errors[0]["error_message"] == "Error (exit code 126)"
```
Comment (pieterlukasse) [inline test/test_argo_engine.py:788]:
```suggestion
```

**Review by pieterlukasse — APPROVED** [review]

**Pull request merged**